app/models/tenant.rb

```diff
@@ -155,6 +155,18 @@ def login_logo?
     !!login_logo_file_name
   end
 
+  def visible_domains
+    MiqAeDomain.where(:tenant_id => ancestor_ids.append(id)).order('priority DESC')
+  end
+
+  def enabled_domains
+    MiqAeDomain.where(:tenant_id => ancestor_ids.append(id), :enabled => true).order('priority DESC')
+  end
+
+  def editable_domains
+    MiqAeDomain.where(:tenant_id => id, :system => false).order('priority DESC')
```

Comment:
I'm fine with doing this for now, I think we'll have to change it once we decide on #4425... I'd like your :tomato: throwing on that PR so we can easily change this PR's code to use that mechanism whenever it lands.